pageserver/src/tenant/timeline/walreceiver/connection_manager.rs

```diff
@@ -606,7 +676,34 @@ impl ConnectionManagerState {
     }
 
     /// Adds another broker timeline into the state, if its more recent than the one already added there for the same key.
-    fn register_timeline_update(&mut self, timeline_update: SafekeeperTimelineInfo) {
+    fn register_timeline_update(&mut self, typed_msg: TypedMessage) {
+        let mut is_discovery = false;
+        let timeline_update = match typed_msg.r#type() {
+            MessageType::SafekeeperTimelineInfo => {
+                let info = typed_msg
+                    .safekeeper_timeline_info
+                    .expect("proto type mismatch from broker message");
+                SafekeeperDiscoveryResponse {
+                    safekeeper_id: info.safekeeper_id,
+                    tenant_timeline_id: info.tenant_timeline_id,
+                    commit_lsn: info.commit_lsn,
+                    safekeeper_connstr: info.safekeeper_connstr,
+                    availability_zone: info.availability_zone,
+                }
+            }
+            MessageType::SafekeeperDiscoveryResponse => {
+                is_discovery = true;
+                typed_msg
+                    .safekeeper_discovery_response
+                    .expect("proto type mismatch from broker message")
+            }
+            _ => {
+                // unexpected message
+                return;
+            }
+        };
+
+        // TODO: make a separate metric for discovery requests
         WALRECEIVER_BROKER_UPDATES.inc();
```

Comment:
Forgotten TODO or planning to do this in a follow-up? If so, what's the ticket for the follow-up?

Reply:
Forgot to remove it, will do.